scripts/rabbitmq-env

```diff
@@ -19,334 +19,510 @@ if [ "$RABBITMQ_ENV_LOADED" = 1 ]; then
+{
+    # Note: default value is 'db'
+    [ -n "$SCHEDULER_BIND_TYPE" ] || SCHEDULER_BIND_TYPE='db'
+    [ -n "$RABBITMQ_SCHEDULER_BIND_TYPE" ] || RABBITMQ_SCHEDULER_BIND_TYPE="$SCHEDULER_BIND_TYPE"
+}
+
+_rmq_env_config_distribution_buffer_arg()
+{
+    # Note: default value is 128000
+    [ -n "$DISTRIBUTION_BUFFER_SIZE" ] || DISTRIBUTION_BUFFER_SIZE=128000
+    [ -n "$RABBITMQ_DISTRIBUTION_BUFFER_SIZE" ] || RABBITMQ_DISTRIBUTION_BUFFER_SIZE="$DISTRIBUTION_BUFFER_SIZE"
+}
+
+_rmq_env_config_server_erl_args()
+{
+    _rmq_env_config_process_limit_arg
+    _rmq_env_config_atom_limit_arg
+    _rmq_env_config_scheduler_bind_type_arg
+    _rmq_env_config_distribution_buffer_arg
+
+    # Note: default value is
+    #   +P $RABBITMQ_MAX_NUMBER_OF_PROCESSES
+    #   +t $RABBITMQ_MAX_NUMBER_OF_ATOMS
+    #   +stbt $RABBITMQ_SCHEDULER_BIND_TYPE
+    #   +zdbbl $RABBITMQ_DISTRIBUTION_BUFFER_SIZE
+    [ -n "$SERVER_ERL_ARGS" ] || SERVER_ERL_ARGS=" +P $RABBITMQ_MAX_NUMBER_OF_PROCESSES +t $RABBITMQ_MAX_NUMBER_OF_ATOMS +stbt $RABBITMQ_SCHEDULER_BIND_TYPE +zdbbl $RABBITMQ_DISTRIBUTION_BUFFER_SIZE "
+    [ -n "$RABBITMQ_SERVER_ERL_ARGS" ] || RABBITMQ_SERVER_ERL_ARGS="$SERVER_ERL_ARGS"
+}
+
+_rmq_env_config_start_args()
+{
+    [ -n "$RABBITMQ_BOOT_MODULE" ] || RABBITMQ_BOOT_MODULE="$BOOT_MODULE"
+    [ -n "$RABBITMQ_ALLOW_INPUT" ] || _rmq_env_noinput_arg='-noinput'
+    [ -n "$RABBITMQ_NODE_ONLY" ] || _rmq_env_node_only_arg="-s $RABBITMQ_BOOT_MODULE boot"
+    [ -n "$_rmq_env_noinput_arg" ] && [ -n "${_rmq_env_node_only_arg}" ] && _rmq_env_maybe_space=' '
+
+    # shellcheck disable=SC2034
+    _rmq_env_start_rabbit="${_rmq_env_noinput_arg}${_rmq_env_maybe_space}${_rmq_env_node_only_arg}"
+
+    unset _rmq_env_noinput_arg
+    unset _rmq_env_node_only_arg
+    unset _rmq_env_maybe_space
+}
+
+_rmq_env_config_alloc_args()
+{
+    # The default allocation strategy RabbitMQ is using was introduced
+    # in Erlang/OTP 20.2.3. Earlier Erlang versions fail to start with
+    # this configuration. We therefore need to ensure that erl accepts
+    # these values before we can use them.
+    #
+    # The defaults are meant to reduce RabbitMQ's memory usage and help
+    # it reclaim memory at the cost of a slight decrease in performance
+    # (due to an increase in memory operations). These defaults can be
+    # overriden using the RABBITMQ_SERVER_ERL_ARGS variable.
+    # shellcheck disable=SC2034
+    #
+    # Note: the default value is
+    #   +MBas ageffcbf
+    #   +MHas ageffcbf
+    #   +MBlmbcs 512
+    #   +MHlmbcs 512
+    #   +MMmcs 30
+    [ -n "$ALLOC_ARGS" ] || ALLOC_ARGS="+MBas ageffcbf +MHas ageffcbf +MBlmbcs 512 +MHlmbcs 512 +MMmcs 30"
+    [ -n "$RABBITMQ_ALLOC_ARGS" ] || RABBITMQ_ALLOC_ARGS="$ALLOC_ARGS"
+}
+
+_rmq_env_config_mnesia_dir()
+{
+    [ -n "$RABBITMQ_MNESIA_DIR" ] || RABBITMQ_MNESIA_DIR="$MNESIA_DIR"
+    [ -n "$RABBITMQ_MNESIA_DIR" ] || RABBITMQ_MNESIA_DIR="$RABBITMQ_MNESIA_BASE}/$RABBITMQ_NODENAME"
```

Comment:
I've removed them when they are not needed. You only need to use curly braces to disambiguate the variable name from another character that is a valid variable character, or when you're using the `%` or `#` trick to do string manipulation.

Neither of these cases need curly braces:

```
echo "$RABBITMQ_MNESIA_DIR foobarbaz"
echo "$RABBITMQ_MNESIA_DIR-foobarbaz"
```

This case does because `RABBITMQ_MNESIA_DIRfoobarbaz` is a valid variable name:

```
echo "${RABBITMQ_MNESIA_DIR}foobarbaz"
```